src/lib/y2network/interface.rb

```diff
@@ -28,14 +28,46 @@ class Interface
     attr_accessor :description
     # @return [Symbol] Interface type
     attr_accessor :type
+    attr_reader :configured
+    attr_reader :hardware
+
+    # Shortcuts for accessing interfaces' ifcfg options
+    ["STARTMODE", "BOOTPROTO"].each do |ifcfg_option|
+      method_name = ifcfg_option.downcase
+
+      define_method method_name do
+        # when switching to new backend we need as much guards as possible
+        if !configured || config.nil? || config.empty?
+          raise "Trying to read configuration of an unconfigured interface #{@name}"
+        end
+
+        config[ifcfg_option]
+      end
+    end
 
     # Constructor
     #
+    # For now until we start to use descendants of interface class, name can be empty.
+    # In such case the system is checked if the device is physically present, but
+    # unconfigured - which is valid. If the name is not provided and the device is not
+    # physical one, an exception is raised.
+    #
     # @param name [String] Interface name (e.g., "eth0")
     def initialize(name, type: InterfaceType::ETHERNET)
       @name = name
       @description = ""
       @type = type
+      # @hardware and @name should not change during life of the object
+      @hardware = Hwinfo.new(name: name)
+
+      if !(name.nil? || name.empty?)
+        @name = name
```

Comment:
This assignment is already done a few lines above.

Reply:
ah yeah  .... mistake from rebasing :-/

Reply:
cleaned